README.md
# Atomize fork

[![Version](https://vsmarketplacebadge.apphb.com/version/anpleenko.atomize-fork.svg)](https://marketplace.visualstudio.com/items?itemName=anpleenko.atomize-fork)

A detailed and accurate Atom One Dark theme

<img src="https://raw.githubusercontent.com/emroussel/atomize/main/images/screenshot.png" alt="Screenshot of VS Code with Atomize" />

To get the icons in the screenshot above and an experience closer to Atom, check out my [Atom Icons theme](https://github.com/emroussel/atom-icons).

**Note that if you want the same syntax highlighting as Atom One Dark, you will need to turn off semantic highlighting in your VS Code settings:**

```json
"editor.semanticHighlighting.enabled": false
```

## Contribution

I have mostly used this theme with JavaScript and other web technologies.

If you'd like me to add support for other languages, or notice a bug/discrepancy with Atom's One Dark theme, feel free to open an issue or pull request on this repo.

When contributing changes, make sure that there is an example of those changes in a file in `/demo`. Files in that folder are used for manually testing syntax highlighting changes. It's not meant to be functional code (although it should be syntaxically correct) but instead to showcase different code patterns of languages supported by this theme. To make sure changes don't break existing highlighting, open and glance at these files in VS Code.

## Motivation

I've always loved the UI and UX of Atom, but prefer the speed and reliability of VS Code. After looking for a while, I couldn't find any VS Code theme that accurately replicated Atom One Dark's interface and syntax highlighting, so I made this one.

## Next steps

Here are some settings I use to make VS Code more minimalistic, so that I can focus on what's important. It also makes the experience more similar to Atom.

```json
{
  "breadcrumbs.enabled": false,
  "editor.cursorBlinking": "blink",
  "editor.folding": false,
  "editor.hideCursorInOverviewRuler": true,
  "editor.minimap.enabled": false,
  "editor.occurrencesHighlight": false,
  "editor.renderIndentGuides": false,
  "editor.roundedSelection": false,
  "editor.selectionHighlight": false,
  "editor.scrollBeyondLastLine": false,
  "explorer.decorations.colors": false,
  "explorer.openEditors.visible": 0,
  "window.zoomLevel": 0,
  "workbench.activityBar.visible": false,
  "workbench.editor.showIcons": false,
  "workbench.startupEditor": "none",
  "workbench.tree.renderIndentGuides": "none"
}
```

**Note:** This will hide the activity bar, so you will need to memorize a few shortcuts (they will be different if you changed the defaults):

- Show explorer: ⇧+ ⌘ + E
- Show search: ⇧+ ⌘ + F
- Show extensions: ⇧+ ⌘ + X

## Credits

This theme is heavily inspired from Atom's [One Dark Syntax theme](https://github.com/atom/one-dark-syntax).

## License

MIT


CHANGELOG.md
# Change Log

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## 2.0.2 (June 10, 2022)

- Add bracket match theme colors

## 2.0.1 (March 9, 2022)

- Correct the scopes for CSS properties

## 2.0.0 (November 24, 2021)

- Revamp syntax highlighting to be more generic and easier to maintain
- Add demo folder with code examples to simplify manual testing
- Drop support for VS Code versions below 1.48.0

## 1.5.5 (April 24, 2021)

- Fix list drop color and remove other deprecated drop keys

## 1.5.4 (March 5, 2021)

- Add new quickInput.list.focusBackground color

## 1.5.3 (November 18, 2020)

- Mark enums as constants
- Update wording in README

## 1.5.2 (August 24, 2020)

- Add highlighting for C++ namespaces and properties

## 1.5.1 (August 3, 2020)

- Fix color of C++ directives

## 1.5.0 (June 23, 2020)

- Add selection highlight colors
- Update recommended vscode settings in README
- Update dev dependencies
- Add activityBar.dropBorder and panel.dropBorder

## 1.4.5 (April 22, 2020)

- Add much better support for .jsx and .tsx files
- Fix color of commas in graphql arguments
- Add git colors to overview ruler

## 1.4.4 (March 12, 2020)

- Add shadow on widgets for better contrast
- Make color of optional chaining operator same as dot
- Fix nested curly braces color in graphql
- Add note about turning off semantic highlighting

## 1.4.3 (February 2, 2020)

- Change color of empty editor for better contrast
- Fix some GraphQL query syntax
- Make booleans in JSON same color as in JS
- Add scripts to format json and md with prettier
- Add Github action workflow to lint on push and PRs

## 1.4.2 (October 27, 2019)

- Format package.json with prettier
- Format CHANGELOG with prettier
- Add new screenshot to match VS Code's new icons
- Fix colors of constants on built in classes
- Add breadcrumbs and indent guide settings to README

## 1.4.1 (July 12, 2019)

- Make remote development icon consistent with other icons

## 1.4.0 (June 2, 2019)

- Add listFilterWidget colors
- Fix color of peek view for errors
- Add right color on warning swiggles
- Fix colors of input validations
- Color numbered lists in markdown
- Add support for TOML
- Clean up colors in peek view
- Add dropdown colors for Windows

## 1.3.1 (April 13, 2019)

- Improve status bar colors for debugging and prominent items
- Add link to atom icons in README
- Compress screenshot in README
- Make foreground color of notifications consistent with lists
- Improve block quote colors in markdown

## 1.3.0 (March 20, 2019)

- Improve git support and add more color contrast in diffs
- Simplify and fix a few syntax colors
- Make small wording improvement in README
- Fix indentation in package.json

## 1.2.0

- Improve Python syntax highlighting support
- Fix and standardize some settings colors

## 1.1.1

- Update description of theme
- Fix some highlighting colors
- Update screenshot

## 1.1.0

- Add color for ignored files
- Fix attribute color in derivative html
- Fix a few small color discrepancies with Atom

## 1.0.7

- Add VS Code Marketplace icon

## 1.0.6

- Make sure all JS keywords are the right color

## 1.0.5

- Update name of `modifiedItemIndicator`
- Remove color on links in code
- Fix color of JS delete

## 1.0.2

- Added more representative screenshot
- Added MIT license
- Updated readme

## 1.0.1

- Compressed image in readme

## 1.0.0

- Initial release


demo/react.js
import React, { Component, useState } from 'react';
import PropTypes from 'prop-types';

export const FunctionalComponent = React.memo((props) => {
  const { title } = props;
  const [count, setCount] = useState(0);

  const increment = () => {
    setCount((prevCount) => prevCount + 1);
  };

  return (
    <div className="class">
      <h1>{title}</h1>
      <p>{count}</p>
      <button type="button" onClick={increment}>
        Increment
      </button>
    </div>
  );
});

FunctionalComponent.propTypes = {
  title: PropTypes.string.isRequired,
};

export class ClassComponent extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  increment() {
    this.setState((prevState) => ({ count: prevState.count + 1 }));
  }

  render() {
    const { title } = this.props;
    const { count } = this.state;

    return (
      <div className="class">
        <h1>{title}</h1>
        <p>{count}</p>
        <button type="button" onClick={this.increment}>
          Increment
        </button>
      </div>
    );
  }
}


demo/javascript.js
// This is a comment
import * as module1 from "module1";
import module2, { module3 } from "module2And3";
const fetch = require("fetch");

/* This is another comment */

const SOME_CONSTANT = "SOME_CONSTANT";
let object = {
  string: "String",
  number: 123,
  float: 123.321,
  boolean: true,
  null: null,
  undefined: undefined,
  array: ["123", "456"],
  date: new Date(),
  function: (array = null) =>
    Array.isArray(array) ? array.reverse() : undefined,
  [module2.dynamicKey]: module2.value,
};

object.function(object.array);

console.log(`Constant: ${SOME_CONSTANT}`);

let num;

if (process.env.NODE_ENV === "production") {
  for (const i = 0; i < module1.some.nested.array.length; i++) {
    if (
      module1.some.nested.array[i] < object.number ||
      module1.some.nested.array[i] - num > object.number
    ) {
      num += i;
      break;
    } else if (module1.some.nested.array[i] === object.number && !num) {
      num -= 1;
    } else {
      for (const property in object) {
        num =
          typeof property === "number" ? Math.floor(num + property) : num + 3;
      }
    }
  }
} else {
  num = Object.keys(object)
    .filter(Boolean)
    .reduce((acc, value) => [...acc, value], []).length;
}

let data = {};

try {
  async function fetchSomething({ url, options: optionsAlias }) {
    const res = await fetch(url, { options: optionsAlias });

    res.json().then((resData) => {
      data = resData?.maybe?.some?.property ?? { default: "default" };
    });

    if (!data) {
      throw new Error("Oh no");
    }
  }

  fetchSomething({ url: "https://123.com" });
} catch (error) {
  if (error instanceof Error) {
    return error;
  }
}

class MyClass extends module3 {
  property = "some property";

  static somethingStatic(value) {
    if (typeof value == "string") {
      return value.test(/^(^https?:\/\/.*)|(^mailto:.*)|(^tel:.*)/i);
    }

    return !!value;
  }

  constructor() {
    super();
  }

  someMethod(...params) {
    (params || []).forEach((param) => {
      this.property += param;
    });
  }
}

MyClass.somethingStatic();

var instance = new MyClass();

instance.someMethod(data);

export default num;

module.exports = {};


demo/typescript.ts
// This is a comment
import * as module1 from 'module1';
import module2, { module3 } from 'module2And3';
const fetch = require('fetch');

/* This is another comment */

const SOME_CONSTANT = 'SOME_CONSTANT';
let object = {
  string: 'String',
  number: 123,
  float: 123.321,
  boolean: true,
  null: null,
  undefined: undefined,
  array: ['123', '456'],
  date: new Date(),
  function: (array: Array<any> = null) => (Array.isArray(array) ? array.reverse() : undefined),
  [module2.dynamicKey]: module2.value,
};

object.function(object.array);

console.log(`Constant: ${SOME_CONSTANT}`);

let num;

if (process.env.NODE_ENV === 'production') {
  for (const i = 0; i < module1.some.nested.array.length; i++) {
    if (module1.some.nested.array[i] < object.number || module1.some.nested.array[i] - num > object.number) {
      num += i;
      break;
    } else if (module1.some.nested.array[i] === object.number && !num) {
      num -= 1;
    } else {
      for (const property in object) {
        num = typeof property === 'number' ? num + property : num + 3;
      }
    }
  }
} else {
  num = Object.keys(object)
    .filter(Boolean)
    .reduce<string[]>((acc, value) => [...acc, value], []).length;
}

let data = {};

interface FetchOptions {
  anOption?: number | null | undefined;
}

type SomeType = keyof FetchOptions;

type Partial<T> = {
  [P in keyof T]?: T[P];
};

type PartialFetchOptions = Partial<FetchOptions>;

try {
  async function fetchSomething({
    url,
    options: optionsAlias,
  }: {
    url: string;
    options?: PartialFetchOptions & { option1: () => SomeType };
  }) {
    const res = await fetch(url, { options: optionsAlias });

    res.json().then((resData) => {
      data = resData?.maybe?.some?.property ?? { default: 'default' };
    });

    if (!data) {
      throw new Error('Oh no');
    }
  }

  fetchSomething({ url: 'https://123.com' });
} catch (error) {
  if (error instanceof Error) {
    return error;
  }
}

enum MyEnum {
  option1 = 1,
  option2,
  option3,
}

class MyClass extends module3 implements module2 {
  private property = 'some private property';
  public publicProperty = 'some public property';

  static somethingStatic(value?: string | number): MyEnum {
    if (!!value.test(/^(^https?:\/\/.*)|(^mailto:.*)|(^tel:.*)/i)) {
      return MyEnum.option1;
    }
  }

  constructor() {
    super();
  }

  someMethod(...params): void {
    (params || []).forEach((param) => {
      this.property += param;
    });
  }
}

MyClass.somethingStatic();

var instance = new MyClass();

instance.someMethod(data);

export default num;


demo/python.py
# This is a comment
from module.first_module import module1, some_decorator
from class_module import AClass as SomeClass

"""
Description  
"""
def some_function(arg=None):
  if not arg:
    raise Exception('Arg is {}'.format(arg))
  elif arg is 'some arbitrary value' or \
    arg is 'some other value with escaping \'':
    return False
  else:
    return True

async def calculate(a, b, c, d, e):
    return await module1.calculate(a + b - c * d / e)

SOME_CONSTANT = 'SOME_CONSTANT'
dictionary = {
  'string': 'String',
  'int': 123,
  'float': 123.123,
  'boolean': True,
  'none': None,
  'array': ['123', '456'],
  'function': some_function
}
array_numbers = [int(value) if value >= 0 and value <= 1000 else 0 for value in dictionary.array]

print('Constant: ' + SOME_CONSTANT)

try:
  some_function(0)
except Exception as excep:
  print(excep)
  pass

class MyClass(SomeClass):
  property = 'some property'

  def __init__(self, new_property):
    self.property = new_property if new_property else module1.PROPERTY_CONSTANT

  @some_decorator(arg=10)
  def some_method(self):
    values1 = (1, 2)
    values2 = {'c': 1, 'd': 2, 'e': 3}
    return calculate(*values1, **values2)
